lib/manageiq/session/mem_cache_store_adapter.rb

```diff
@@ -13,7 +13,8 @@ def session_options
           :key             => "_vmdb_session",
           :memcache_server => MiqMemcached.server_address,
           :namespace       => "MIQ:VMDB",
-          :value_max_bytes => 10.megabytes,
+          :pool_size       => 10,
```

Comment:
Use the default of 10 for our normal session requests

Reply:
If the default has 10 why do we specify it here again?

Reply:
Sorry, I should have said the "suggested value" of 10.  

We need to specify something for the connection pool, such as the `pool_size` or `pool_timeout` to get it to use a pool, see [this](https://github.com/petergoldstein/dalli/blob/v3.1.6/lib/rack/session/dalli.rb#L129-L136).  They specify a "suggested" value of 10 [here](https://github.com/petergoldstein/dalli/blob/v3.1.6/lib/rack/session/dalli.rb#L44-L57), I didn't look to see if it's the default but even if it is, the default is further away than in dalli, such as in something like Rack session, so it makes sense to specify the value

Reply:
As discussed, I wonder if this should be `Puma.threads` or somehow interrelate the two.

Reply:
> As discussed, I wonder if this should be Puma.threads (pseudocode) or somehow interrelate the two.

I looked and don't think we can do it.
* We don't know the runtime number of workers/max threads at this point as the web server isn't yet started, so you need to check the boot configuration of puma 
* puma can be run in different ways.  If we ever change it, we might need to change this introspection as it can be configured:
  * from `config/puma.rb` (current way)
  * via command line using the puma executable
  * with a different config file: `rails server --config=path/to/another_file`
  * ... ?

It looks like the default from connection_pool gem is a size 5 pool:
https://github.com/mperham/connection_pool/blob/02fb930fc988b9a9f45011d7413231627a24d497/lib/connection_pool.rb#L39

So, before this change: https://github.com/ManageIQ/manageiq/commit/49c1132983aa346f94f821964f7eb3ea7c46ff7e#diff-395a4ac087f9efbc29d305a5da65853725c23c1ec1b2556bf38875abf1bf6632R14, I believe we were using `threadsafe` in dalli as we didn't specify a value. `Dalli.new` defaults to `true` if `threadsafe` isn't specified.  Additionally, because we specified neither `pool_size` nor `pool_timeout`, we weren't using a connection pool.

With that change, we went from `threadsafe` with no pool to connection pool but a size of 1.

So, if we want to use a connection pool we need to specify one of the pool settings: `pool_size` or `pool_timeout` to get it to use the [connection pool](https://github.com/petergoldstein/dalli/blob/v3.1.6/lib/rack/session/dalli.rb?rgh-link-date=2022-02-25T20%3A13%3A30Z#L129-L136).  I don't know if 5 is adequate as main worker loop may also use a connection so I think we need at least 6.  Therefore, I took the easy option of saying 10.

Reply:
Sounds good to me. Thanks for the thoughtful detail.